pkg/etcd/metadata_controller.go

```diff
@@ -108,6 +108,18 @@ func (m *metadataHandler) handleSelf(node *v1.Node) (*v1.Node, error) {
 			node.Labels = map[string]string{}
 		}
 
+		findEtcdStatusCondition := -1
+		for i, v := range node.Status.Conditions {
+			if v.Type == etcdStatusType {
+				findEtcdStatusCondition = i
+			}
+		}
+
+		// remove etcd status condition if exists
+		if findEtcdStatusCondition >= 0 {
+			node.Status.Conditions = append(node.Status.Conditions[:findEtcdStatusCondition], node.Status.Conditions[findEtcdStatusCondition+1:]...)
+		}
```

Comment:
Use util.GetNodeCondition, same as you did previously.
https://github.com/k3s-io/k3s/blob/3190a5faa28d7a0d428c756d67adcab7eb11e6a5/pkg/etcd/etcd.go#L1186